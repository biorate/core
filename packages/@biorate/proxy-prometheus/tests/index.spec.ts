import { expect } from 'chai';
import { root, Root } from './__mocks__';

describe('@biorate/proxy', function () {
  this.timeout(30e3);

  it('connection', async () => {
    await root.$run();
  });

  it('write', (done) => {
    const socket = Root.connect();
    socket.write('Hello');
    socket.on('data', (data) => {
      expect(data.toString()).toMatchSnapshot();
      done();
    });
  });

  it('metrics', async () => {
    const result = await root.prometheus.registry.metrics();
    expect(result.includes('proxy_connector_write_bytes')).to.be.a('boolean').equal(true);
    expect(result.includes('proxy_connector_read_bytes')).to.be.a('boolean').equal(true);
    expect(result.includes('proxy_connector_active')).to.be.a('boolean').equal(true);
  });
});
