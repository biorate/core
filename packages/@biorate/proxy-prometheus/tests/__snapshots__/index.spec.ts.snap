// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@biorate/proxy write 1`] = `"Hello world!"`;
